commons/all/src/main/java/org/infinispan/commons/marshall/PersistenceContextInitializer.java

```diff
@@ -23,6 +24,7 @@
             CounterConfiguration.class,
             CounterType.class,
             Storage.class,
+            XidImpl.class
```

Comment:
When do Xids need to be persisted?

If they don't, then `XidImpl` should continue to utilise an Externalizer until the global marshaller has been refactored (Infinispan 12 now).

Reply:
hmmm... I thought we were moving everything to protostream... aren't we? 
I guess I can rollback this change... And `org.infinispan.transaction.xa.recovery.InDoubtTxInfo` too

Reply:
> I thought we were moving everything to protostream... aren't we?

We are, however there are two contexts for marshalling. The persistence context and the global context. The persistence context is tightly controlled and should only include classes that need to be persisted. We have stronger guarantees over this context at the moment, as we need to ensure store compatibility across majors. This is what is currently implemented in ProtoStream, hence why you added `XidImpl` to the `PersistenceContextInitializer`.

The global context is for general marshalling required for inter-node communication etc. This hasn't been converted to be protostream yet, well not on master at least, so if you only require marshalling for this process you continue to use Externalizers and in my conversion branch I will handle it. Once my work is resumed for 12, there will be a GlobalContextInitializer which you would add `XidImpl` to.

Reply:
> And `org.infinispan.transaction.xa.recovery.InDoubtTxInfo` too

Yes please